src/toil/wdl/wdltoil.py

```diff
@@ -1979,15 +1984,23 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
 
                 per_part_size = convert_units(part_size, part_suffix)
                 total_bytes += per_part_size
-                if specified_mount_point is not None:
-                    if mount_spec.get(specified_mount_point) is not None:
+                if mount_spec.get(specified_mount_point) is not None:
+                    if specified_mount_point is not None:
                         # raise an error as all mount points must be unique
                         raise ValueError(f"Could not parse disks = {disks_spec} because the mount point {specified_mount_point} is specified multiple times")
-
-                    # TODO: we always ignore the disk type and assume we have the right one.
-                    if specified_mount_point != "local-disk":
-                        # Don't mount local-disk. This isn't in the spec, but is carried over from cromwell
-                        mount_spec[specified_mount_point] = int(per_part_size)
+                    else:
+                        if mount_spec.get(specified_mount_point) is not None:
```

Comment:
This condition is inside another `if` with the same condition, so it will always be true.